manage_breast_screening/mammograms/tests/forms/test_mastectomy_or_lumpectomy_history_form.py

```diff
@@ -0,0 +1,347 @@
+        )
+        assert obj.surgery_other_reason_details == ""
+        assert obj.additional_details == ""
+
+    @pytest.mark.parametrize(
+        "data",
+        [
+            {
+                "right_breast_procedure": MastectomyOrLumpectomyHistoryItem.Procedure.MASTECTOMY_TISSUE_REMAINING,
+                "left_breast_procedure": MastectomyOrLumpectomyHistoryItem.Procedure.MASTECTOMY_NO_TISSUE_REMAINING,
+                "right_breast_other_surgery": [
+                    MastectomyOrLumpectomyHistoryItem.Surgery.RECONSTRUCTION,
+                    MastectomyOrLumpectomyHistoryItem.Surgery.SYMMETRISATION,
+                ],
+                "left_breast_other_surgery": [
+                    MastectomyOrLumpectomyHistoryItem.Surgery.NO_OTHER_SURGERY,
+                ],
+                "surgery_reason": MastectomyOrLumpectomyHistoryItem.SurgeryReason.RISK_REDUCTION,
+            },
+            {
+                "right_breast_procedure": MastectomyOrLumpectomyHistoryItem.Procedure.NO_PROCEDURE,
+                "left_breast_procedure": MastectomyOrLumpectomyHistoryItem.Procedure.LUMPECTOMY,
+                "right_breast_other_surgery": [
+                    MastectomyOrLumpectomyHistoryItem.Surgery.NO_OTHER_SURGERY
+                ],
+                "left_breast_other_surgery": [
+                    MastectomyOrLumpectomyHistoryItem.Surgery.RECONSTRUCTION,
+                    MastectomyOrLumpectomyHistoryItem.Surgery.SYMMETRISATION,
+                ],
+                "year_of_surgery": datetime.date.today().year - 80,
+                "surgery_reason": MastectomyOrLumpectomyHistoryItem.SurgeryReason.GENDER_AFFIRMATION,
+            },
+            {
+                "right_breast_procedure": MastectomyOrLumpectomyHistoryItem.Procedure.LUMPECTOMY,
+                "left_breast_procedure": MastectomyOrLumpectomyHistoryItem.Procedure.LUMPECTOMY,
+                "right_breast_other_surgery": [
+                    MastectomyOrLumpectomyHistoryItem.Surgery.RECONSTRUCTION
+                ],
+                "left_breast_other_surgery": [
+                    MastectomyOrLumpectomyHistoryItem.Surgery.RECONSTRUCTION
+                ],
+                "year_of_surgery": datetime.date.today().year,
+                "surgery_reason": MastectomyOrLumpectomyHistoryItem.SurgeryReason.OTHER_REASON,
+                "surgery_other_reason_details": "surgery other details",
+                "additional_details": "additional details",
+            },
+        ],
+    )
+    def test_success(self, clinical_user, data):
+        appointment = AppointmentFactory()
+        request = RequestFactory().get("/test-form")
+        request.user = clinical_user
+
+        form = MastectomyOrLumpectomyHistoryForm(
+            QueryDict(urlencode(data, doseq=True)),
+            participant=appointment.participant,
+        )
+
+        assert form.is_valid()
+
+        existing_log_count = AuditLog.objects.count()
```

Comment:
This test is doing quite a lot of verification... I'm wondering if it's worth doing quite so much querying of the database state in our tests as it's going to slow down the runtime of the test suite. 

I previously added a similar test to this audit log bit for something else, but we decided in review to drop it, and spoke about adding some kind of static analysis check instead. I haven't had a go at that yet but think we should probably leave it out here as well.